includes/REST/StoreSettingController.php

```diff
@@ -41,10 +41,48 @@ public function register_routes() {
 					'callback'            => [ $this, 'get_settings' ],
 					'permission_callback' => [ $this, 'get_settings_permission_callback' ],
 				],
+                [
+                    'methods'             => WP_REST_Server::EDITABLE,
+                    'callback'            => [ $this, 'update_settings' ],
+                    'permission_callback' => [ $this, 'get_settings_permission_callback' ],
+                ],
 			]
         );
     }
 
+    /**
+     * Update Store
+     *
+     * @param \WP_REST_Request $request
+     *
+     * @since DOKAN_LITE_SINCE
+     *
+     * @return WP_Error|\WP_REST_Response
+     */
+    public function update_settings( $request ) {
+        $vendor = $this->get_vendor();
+
+        if ( empty( $vendor->get_id() ) ) {
+            return new WP_Error( 'no_store_found', __( 'No vendor found', 'dokan-lite' ), [ 'status' => 404 ] );
+        }
+
+        $params   = $request->get_params();
+        $store_id = dokan()->vendor->update( $vendor->get_id(), $params );
+
+        if ( is_wp_error( $store_id ) ) {
+            return new WP_Error( $store_id->get_error_code(), $store_id->get_error_message() );
+        }
+
+        $store = dokan()->vendor->get( $store_id );
+
+        do_action( 'dokan_rest_store_settings_update_store', $store, $request );
```

Comment:
Consider changing the name of the hook name `dokan_rest_store_settings_update_store` to `dokan_rest_store_settings_after_update`